### PR TITLE
Makes the FreeBSD Autotools config file work like Linux

### PR DESCRIPTION
**Pull request by derobins**
Defaults are cc (was: gcc) and f77

### CODE REVIEW

**Comment by derobins** [comment]
We already made these changes in HDF5, so they were easy to port here.

File: release_notes/RELEASE.txt [hunk 1]
```diff
@@ -38,6 +38,18 @@ New features and changes
 ========================
     Configuration:
     -------------
+    - The FreeBSD Autotools config was updated to work like MacOS and Linux
+
+      config/freebsd was updated to try a variety of compilers and the
```
Comment (bmribler) [inline release_notes/RELEASE.txt:43]: Perhaps: "The file config/freebsd" or something...

File: config/freebsd [hunk 2]
```diff
@@ -20,110 +20,121 @@
...
+        fi
+        ;;
+
+     *)
+        echo "No match to get fc_version_info for $F77"
+        ;;
 esac
 
+if test X != "X$fc_version"; then
+    # Get the compiler version numbers
```
Comment (bmribler) [inline config/freebsd:134]: number vs. numbers?

File: config/freebsd [hunk 3]
```diff
@@ -20,110 +20,121 @@
...
+        fc_version_info=`$F77 $FFLAGS -V 2>&1 | grep 'Version' |\
+            sed 's/\(Intel.* Compiler\).*\( Version [a-z0-9\.]*\).*\( Build [0-9]*\)/\1\2\3/'`
+        fc_version="`$F77 $FFLAGS -V 2>&1 |grep '^Intel'`"
+        if test X != "X$fc_version"; then
+            fc_version="`echo $fc_version |sed 's/.*Version \([-a-z0-9\.\-]*\).*/\1/'`"
+        fi
+        ;;
+
+    *f95*)
+        # Figure out which compiler we are using: pgf90 or Absoft f95
```
Comment (bmribler) [inline config/freebsd:102]: I don't see pgf90

**Review by bmribler — APPROVED** [review]

File: config/freebsd (hunk 2 above)
Comment (derobins) [reply config/freebsd:134]: I think they say "numbers" because it's often something like "12.1.0" (major.minor.patch)

File: config/freebsd (hunk 3 above)
Comment (derobins) [reply config/freebsd:102]: It's copied over verbatim from HDF5. Since I plan to bring over PGI support in a day or three, I figured I'd leave it in.

**Review by byrnHDF — APPROVED** [review]

**Pull request merged**